resize-observer/devicepixel3-ref.html

```diff
@@ -0,0 +1,77 @@
+<!doctype html>
+<style>
+  .outer {
+    display: flex;
+    align-items: center;
+    flex-direction: column;
+  }
+  .outer>* {
+    display: block;
+    height: 100px;
+  }
+</style>
+<body>
+  <div class="outer"></div>
+  <script type="module">
+import createPatternDataURL from './create-pattern-data-url.js';
+
+const {patternSize, dataURL} = createPatternDataURL();
+
+/**
+ * Set the pattern's size on this element so that it draws where
+ * 1 pixel in the pattern maps to 1 devicePixel.
+ */
+function setPattern(elem) {
+  const oneDevicePixel = 1 / devicePixelRatio;
```

Comment:
I am always cautious about usage of multiplications by `window.devicePixelRatio`, fractional sizes can lead to surprising rounding errors. I trust that you understand this better than I do.

Reply:
I'm not sure what you mean by "slight different". If you mean the patterns don't align between divs that's expected as the pattern starts at the left most pixel of each div

Reply:
I think this is what can happen.

Example: devicePixelRation = 1.66.
You will set background size to a float.
To display a background image, browser will round background size to an integer.
There is no way for you to know if size got rounded up or down.
patternPixel pattern size might end up not being an exact multiple of background size, so you can't guarantee it'll display correctly.

Reply:
I don't think that can happen. The browser computes the element size (and therefore the background size) in CSS pixels. The background size is also specified in CSS pixels. Because they are the same units they'll both be converted to device pixels in the same way. It would be a background pattern bug for the pattern not to match what's specified

Reply:
I no longer work on Chrome, so I can't consult the rendering team for the details why rendering fractional pixel widths can create scaling artifacts. If you can find a reviewer that agrees with your interpretation what rendering will look like, I am happy to ok this pull request.

Reply:
Aside: Trippy test! It's cool how this shows when MacOS re-computes window-to-device mapping after you stop moving the window.

Is this test testing devicePixelContentBoxSize? If so, do we really need to render pixels to test it, rather than just checking the values in javascript? The javascript approach will be a faster test to run, and won't have a dependency on how css background images paint.

WPT uses a device pixel ratio of 1 ([issue](https://github.com/web-platform-tests/wpt/issues/12725)) so it is strange that firefox is failing. Is their failure just related to devicePixelContentBoxSize with fractional css width/heights? If so, this would also be cleaner to test using javascript.

Reply:
So, @progers explained that browsers do random things with patterns (or at least Chrome does) and while this test doesn't trigger those random things there's no guarantee that won't change in the future. 

So, closing this PR